### PR TITLE
Added function that checks if a string is an isogram

### PR DESCRIPTION
**Pull request by M3talM0nk3y**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: strings/check_isogram.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+"""
+wiki: https://en.wikipedia.org/wiki/Heterogram_(literature)#Isograms
+"""
+
+
+def check_isogram(string: str) -> bool:
```
Comment (cclauss) [inline strings/check_isogram.py:6]: This is a slightly more self-documenting name for a function that returns a bool.
```suggestion
def is_isogram(string: str) -> bool:
```

Also, please add a test for `true123`.  Should that return True or raise a ValueError?

File: strings/check_isogram.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    >>> check_isogram('allowance')
+    False
+    """
+    letters = sorted(string)
+
+    for idx, letter in enumerate(letters):
+        if letter == letters[idx - 1]:
+            return False
+
+    return True
```
Comment (cclauss) [inline strings/check_isogram.py:21]:
```suggestion
    return len(letters) == len(set(letters))
```

File: strings/check_isogram.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+if __name__ == "__main__":
+    input_str = input("Enter a string ").strip().lower()
```
Comment (cclauss) [inline strings/check_isogram.py:25]: The `str.lower()` should happen ___in___ the function.

File: strings/check_isogram.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
+"""
+wiki: https://en.wikipedia.org/wiki/Heterogram_(literature)#Isograms
+"""
+
+
+def check_isogram(string: str) -> bool:
+    """
+    An isogram is a word in which no letter is repeated.
+    Examples of isograms are uncopyrightable and ambidextrously.
+    >>> check_isogram('uncopyrightable')
```
Comment (cclauss) [inline strings/check_isogram.py:10]:
```suggestion
    >>> check_isogram('Uncopyrightable')
```

File: strings/check_isogram.py (hunk 1 above)
Comment (M3talM0nk3y) [reply strings/check_isogram.py]: Renamed function and added test for when a string contains 1 or more digits. The latter now raises a ValueError. @cclauss

File: strings/check_isogram.py (hunk 2 above)
Comment (M3talM0nk3y) [reply strings/check_isogram.py]: @cclauss Made suggested changes.

File: strings/check_isogram.py (hunk 3 above)
Comment (M3talM0nk3y) [reply strings/check_isogram.py]: @cclauss Lowercasing string inside function.

**Comment by M3talM0nk3y** [comment]
@cclauss Thanks for your feedback. :)

File: strings/is_isogram.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    >>> is_isogram('allowance')
+    False
+    >>> is_isogram('copy1')
+    Traceback (most recent call last):
+     ...
+    ValueError: String must only contain alphabetic characters.
+    """
+    import re
+
+    if bool(re.search(r"\d", string)):
```
Comment (ChrisO345) [inline strings/is_isogram.py:21]:
```suggestion
    if not all(x.isalpha() for x in string):
```
This reduces the need to import `re` and catches things other non alphabetic characters such as symbols.

**Review by ChrisO345 — CHANGES_REQUESTED** [review]

**Review by ChrisO345 — APPROVED** [review]

**Pull request merged**